napari/layers/shapes/_shapes_mouse_bindings.py

```diff
@@ -178,53 +263,216 @@ def _add_line_rectangle_ellipse(layer, event, data, shape_type):
+
         # Start drawing a path
-        data = np.array([coordinates, coordinates])
-        layer.add(data, shape_type='path')
-        layer.selected_data = {layer.nshapes - 1}
-        layer._value = (layer.nshapes - 1, 1)
-        layer._moving_value = copy(layer._value)
-        layer._is_creating = True
-        layer._set_highlight()
+        initiate_polygon_draw(layer, coordinates)
+        yield
+
+        while event.type == 'mouse_move':
+            polygon_creating(layer, event)
+            yield
+        index = layer._moving_value[0]
+        vertices = layer._data_view.shapes[index].data
+        # If number of vertices is higher than 2, tablet draw mode is assumed and shape is finished upon mouse release
+        if len(vertices) > 2:
+            layer._finish_drawing()
+    else:
+        # This code block is responsible for finishing drawing in mouse draw mode
+        layer._finish_drawing()
+
+
+def add_vertex_to_path(
+    layer: Shapes,
+    event: ReadOnlyWrapper,
+    index: int,
+    coordinates: Tuple[np.float, ...],
+    new_type: Optional[str],
+) -> None:
+    """Adds a vertex to an existing path or polygon and edits the layer view.
+
+    Parameters
+    ----------
+    layer: Shapes
+        Napari shapes layer
+    event: ReadOnlyWrapper
+        A proxy read only wrapper around a mouse event
+    index: int
+        The index of the shape being added, e.g. first shape in the layer has index 0.
+    coordinates: Tuple[np.float, ...]
+        The coordinates of the vertex being added to the shape being drawn in image data space
+    new_type: Optional[str]
+        Type of the shape being added.
+
+    Returns
+    -------
+    None
+    """
+    vertices = layer._data_view.shapes[index].data
+    vertices = np.concatenate((vertices, [coordinates]), axis=0)
+    value = layer.get_value(event.position, world=True)
+    layer._value = (value[0], value[1] + 1)
+    layer._moving_value = copy(layer._value)
+    layer._data_view.edit(index, vertices, new_type=new_type)
+    layer._selected_box = layer.interaction_box(layer.selected_data)
+    layer._last_cursor_position = np.array(event.pos)
+
+
+def polygon_creating(layer: Shapes, event: ReadOnlyWrapper) -> None:
+    """Function providing the functionality of moving the vertex when moving the mouse and adding it if the distance
+    threshold criterium is met.
+
+    Parameters
+    ----------
+    layer: Shapes
+        Napari shapes layer
+    event: ReadOnlyWrapper
+        A proxy read only wrapper around a mouse event
+
+    Returns
+    -------
+    None
+    """
+    if layer._is_creating:
+        coordinates = layer.world_to_data(event.position)
+        move_vertex(layer, coordinates)
+
+        if layer._mode == Mode.ADD_POLYGON_LASSO:
+            index = layer._moving_value[0]
+
+            position_diff = np.linalg.norm(
+                event.pos - layer._last_cursor_position
+            )
+            if position_diff > 10:
```

Comment:
Agree with jni

Reply:
@melonora do you have time to add a preference here? 🙏 Having said that @melonora @Czaki I think it would be ok to merge without that and make an issue about it / fix in followup PR. This is already a great PR adding great functionality.

Reply:
I can add it to the settings, wouldn't be a problem. Sorry I missed this comment initially.

Reply:
Resolved with [0d2e01c](https://github.com/napari/napari/pull/5555/commits/0d2e01c69ab129fc9e399f502e80bacd72412ff6). I will also update the docs PR and PR description.

Reply:
Ah nvm regarding the PR description. I can't update it myself and @LucaMarconato is on holiday at the moment. If you, @Czaki or @jni, could do this, it would be much appreciated. Basically add a line:

Exposing the vertex distance threshold in napari preferences -> experimental -> lasso_vertex_distance.
The wheel cannot be set lower than 1 and not higher than 50. Default value is 10. 

Along with some changes in the text above to already indicate that 10 screen pixels is a setting.

Reply:
@psobolewskiPhD I updated the docs PR, could you have a look please?